gcp/src/test/java/org/apache/iceberg/gcp/gcs/GCSFileIOTest.java

```diff
@@ -100,6 +121,70 @@ public void testDelete() {
         .isZero();
   }
 
+  private String gsUri(String path) {
+    return format("gs://%s/%s", TEST_BUCKET, path);
+  }
+
+  @Test
+  public void testListPrefix() {
+    String prefix = "list/path/";
+    String path1 = prefix + "data1.dat";
+    storage.create(BlobInfo.newBuilder(TEST_BUCKET, path1).build());
+    String path2 = prefix + "data2.dat";
+    storage.create(BlobInfo.newBuilder(TEST_BUCKET, path2).build());
+    String path3 = "list/skip/data3.dat";
+    storage.create(BlobInfo.newBuilder(TEST_BUCKET, path3).build());
+
+    assertThat(StreamSupport.stream(io.listPrefix(gsUri("list/")).spliterator(), false).count())
```

Comment:
sorry I missed this initially, but there's no need to wrap this in a stream. This (and other places) can be simplified to `assertThat(io.listPrefix(gsUri("list/"))).hasSize(3);`

Reply:
I noticed there are also existing tests in that file that do the same thing, could you please update those as well?

Reply:
The intent was to avoid using the FileIO we're testing to do the assertion and use the Google client directly, I assume the same for the existing tests.